proposals/solid-oidc/primer.bs

```diff
@@ -0,0 +1,745 @@
+    "redirect_uris" : ["https://decentphotos.example/callback"],
+    "client_name" : "DecentPhotos",
+    "client_uri" : "https://decentphotos.example/",
+    "logo_uri" : "https://decentphotos.example/logo.png",
+    "tos_uri" : "https://decentphotos.example/tos.html",
+    "scope" : "openid profile offline_access",
+    "grant_types" : ["refresh_token","authorization_code"],
+    "response_types" : ["code"],
+    "default_max_age" : 60000,
+    "require_auth_time" : true
+    }""" .
+```
+
+Notice that the application WebID contains one required triple `?webid solid:oidcRegistration ?jsonRegistration`. This is a JSON representation of an [OIDC Client Registration](https://tools.ietf.org/html/rfc7591#section-2).
+
+#### 8. Validate redirect url with WebID
+
+Check to be sure that the `redierct_uri` value provided in the auth request (`https://decentphotos.example/callback`) is listed in the `redirect_uris` array in the oidcRegistration. If it is not, the OP MUST reject the request with a `403`. In our case, it is, so we may continue.
+
+#### 9. Alice Logs In
+
+The OP should redirect to its login screen. The actual implementation of this is completely up to the OP. A user can log in with her password, a TLS certificate, or any other proven method of authentication. The important thing is that, thanks to the redirect, the control is now out of the hands of the RP and is in complete control of the OP.
+
+#### 10. Generates the id_token
+
+If `id_token` was listed as a response type during the authorization request, the OP generates an id token. The id token is not used to access anything. It is simply to communicate information to the client. This token is a (JSON Web Token)[https://jwt.io/introduction/]. It would look like the following (you can decrypt the token with https://jwt.io):
+
+```
+eyJhbGciOiJFUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiJodHRwczovL2FsaWNlLmNvb2xwb2QuZXhhbXBsZS9wcm9maWxlL2NhcmQjbWUiLCJhdWQiOiJodHRwczovL2RlY2VudHBob3Rvcy5leGFtcGxlL3dlYmlkI3RoaXMiLCJ3ZWJpZCI6Imh0dHBzOi8vYWxpY2UuY29vbHBvZC5leGFtcGxlL3Byb2ZpbGUvY2FyZCNtZSIsImlzcyI6Imh0dHBzOi8vc2VjdXJlYXV0aC5leGFtcGxlIiwianRpIjoiODQ0YTA5NWMtOWNkYi00N2U1LTk1MTAtMWRiYTk4N2MwYTVmIiwiaWF0IjoxNjAzMzg2NDQ4LCJleHAiOjE2MDMzODcwNDh9.T306vT8dmn9gQIMEdG92AM4WRnrhqWZTfDpovwqZ6Zn0mK9yxj0iOVGqXD4CW8-tzDTitNwEGorAo85atL0Oeg
+```
+
+Token Header:
+```
+{
+  "alg": "ES256",
+  "typ": "JWT"
+}
+```
+ - `"alg": "ES256"`: indicates the token was signed using eliptic curve
+ - `"typ": "JWT"`: indicates that this is a JSON web token
+
+Token Body:
+```
+{
+  "sub": "https://alice.coolpod.example/profile/card#me",
+  "aud": "https://decentphotos.example/webid#this",
+  "webid": "https://alice.coolpod.example/profile/card#me",
+  "iss": "https://secureauth.example",
+  "jti": "844a095c-9cdb-47e5-9510-1dba987c0a5f",
+  "iat": 1603370641,
+  "exp": 1603371241
+}
+```
+
+ - `"sub": "https://alice.coolpod.example/profile/card#me"`: The user this token corresponds to. (their WebId)
```

Comment:
So, should I wait for the spec to be updated? What do you think it will change to?

Reply:
That's a good question, and we should discuss it at the next panel meeting. This certainly isn't something that needs to hold up this PR, but it would be good to make sure anyone reading this spec understands that the `sub` claim will not reliably return a WebID.

The main use case includes circumstances where someone has an existing IdP that is in broad use and is able to extend that IdP so that it produces `webid` claim. But in order to not break all the existing (non-Solid) applications that use the IdP, the `sub` claim would need to keep its existing form. That is, the `sub` claim might be `user1234` or, in this case: `alice`.